File: 448-find-all-numbers-disappeared-in-an-array/find-all-numbers-disappeared-in-an-array.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        vector<int> missing;
        unordered_set<int> set_nums;

        for (int num : nums)
        {
            set_nums.insert(num);
        }

        for (int i = 1; i <= nums.size(); i++ )
        {
            if (!set_nums.contains(i))
                missing.push_back(i);
        }
        return missing;
    }
};

File: 448-find-all-numbers-disappeared-in-an-array/find-all-numbers-disappeared-in-an-array.java
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        Set<Integer> setNums = new HashSet<>();

        for(int num : nums)
        {
            setNums.add(num);
        }
        ArrayList<Integer> result = new ArrayList<>();

        for (int i = 1; i <= nums.length; i++){
            if (!setNums.contains(i))
                {
                    result.add(i);
                }
        }
        return result;

    }
}